pallets/portfolio/src/lib.rs

```diff
@@ -329,4 +360,177 @@ impl<T: Trait> Module<T> {
+            AuthorizationData::PortfolioCustody(pid) => pid,
+            // Since this function is only called by the Identity pallet after making this check, this line will never be triggered.
+            // Being defensive here anyway since we might want to expose this function via other forms some day.
+            _ => return Err(Error::<T>::IrrelevantAuthorization.into()),
+        };
+
+        let current_custodian =
+            <PortfolioCustodian>::get(&portfolio_id).unwrap_or(portfolio_id.did);
+
+        <identity::Module<T>>::consume_auth(
+            current_custodian,
+            Signatory::from(new_custodian),
+            auth_id,
+        )?;
+
+        <PortfolioCustodian>::insert(&portfolio_id, new_custodian);
+
+        Self::deposit_event(RawEvent::PortfolioCustodianChanged(
+            new_custodian,
+            portfolio_id,
+            new_custodian,
+        ));
+        Ok(())
+    }
+
+    /// Locks some user tokens so that they can not be used for transfers.
+    /// This is used internally by the settlement engine to prevent users from using the same funds
+    /// in multiple ongoing settlements
+    ///
+    /// # Errors
+    /// * `InsufficientPortfolioBalance` if the portfolio does not have enough free balance to lock
+    fn lock_tokens(
+        portfolio: &PortfolioId,
+        ticker: &Ticker,
+        amount: &T::Balance,
+    ) -> DispatchResult {
+        // 1. Ensure portfolio has enough free balance
+        let balance = Self::portfolio_asset_balances(portfolio, ticker);
+        ensure!(
+            balance
+                .saturating_sub(Self::locked_assets(portfolio, ticker))
+                .checked_sub(&amount)
+                .is_some(),
+            Error::<T>::InsufficientPortfolioBalance
+        );
+
+        // 2. Lock tokens.
+        // Locks are stacked so if there were X tokens already locked, there will now be X + N tokens locked
+        <PortfolioLockedAssets<T>>::mutate(portfolio, ticker, |locked| {
+            *locked = locked.saturating_add(*amount)
+        });
+        Ok(())
+    }
+
+    /// Unlocks some locked tokens of a user.
+    /// Since this is only ever called by the settlement engine,
+    /// it will never be called under circumstances when it has to return an error.
+    /// We are being defensive with the checks anyway.
+    ///
+    /// # Errors
+    /// * `InsufficientTokensLocked` if the portfolio does not have enough locked tokens to unlock
+    fn unlock_tokens(
+        portfolio: &PortfolioId,
+        ticker: &Ticker,
+        amount: &T::Balance,
+    ) -> DispatchResult {
+        // 1. Ensure portfolio has enough locked tokens
+        let locked = Self::locked_assets(portfolio, ticker);
+        ensure!(locked >= *amount, Error::<T>::InsufficientTokensLocked);
+
+        // 2. Unlock tokens. Can not underflow due to above ensure.
+        <PortfolioLockedAssets<T>>::insert(portfolio, ticker, locked - *amount);
```

Comment:
instead of insert `mutate` does makes more sense here

Reply:
I am already fetching the value two lines above. `mutate` causes an extra read that we don't need.